drivers/spi/spi_ll_stm32.c

```diff
@@ -845,6 +835,15 @@ static int spi_stm32_transceive_async(const struct device *dev,
 				      spi_callback_t cb,
 				      void *userdata)
 {
+#ifdef CONFIG_SPI_STM32_DMA
+	struct spi_stm32_data *data = dev->data;
+
+	if ((data->dma_tx.dma_dev != NULL)
+	 && (data->dma_rx.dma_dev != NULL)) {
+		return transceive_dma(dev, config, tx_bufs, rx_bufs,
```

Comment:
The `transceive_dma` function calls `wait_dma_rx_tx_done`, so it waits for transfer to end, so it is not an asynchronous mode. It looks like, it is a synchronous mode with calling the callback at the end of transfer.

Reply:
Change the transceive_dma when the ASYNC mode is activated: behaves like the transceive() does.